conda/cli/activate.py

```diff
@@ -150,13 +151,18 @@ def main():
         if sys.argv[2] == 'root':
             # no need to check root env and try to install a symlink there
             sys.exit(0)
-        binpath = binpath_from_arg(sys.argv[2])  # this should throw an error and exit if the env or path can't be found.
+
+        # this should throw an error and exit if the env or path can't be found.
+        binpath = binpath_from_arg(sys.argv[2])
+
         # Make sure an env always has the conda symlink
         try:
             conda.install.symlink_conda(binpath[0], conda.config.root_dir, find_parent_shell())
         except (IOError, OSError) as e:
             if e.errno == errno.EPERM or e.errno == errno.EACCES:
-                sys.exit("Cannot activate environment {}, do not have write access to write conda symlink".format(sys.argv[2]))
+                msg = "Cannot activate environment {}, ".format(sys.argv[2])
+                msg += "not have write access to conda symlink"
```

Comment:
This string concat is bad for performance.  End up making three copies when you only need one.  Do this instead

    msg = ("Cannot activate environment {0}, "
                "not have write access to conda symlink".format(sys.argv[2]))

Can also do this

    msg = ("Cannot activate environment {0}, not have write access to conda symlink"
                .format(sys.argv[2]))

Also note making the index `{0}` explicit.  Keeps compatibility with python 2.6 (even though we don't explicitly support it) and it also performs *much* faster.